lib/phoenix/socket/transport.ex

```diff
@@ -593,8 +593,13 @@ defmodule Phoenix.Socket.Transport do
 
   defp origin_allowed?({module, function, arguments}, uri, _endpoint, _conn),
     do: apply(module, function, [uri | arguments])
-  defp origin_allowed?(:conn, uri, _endpoint, %Plug.Conn{} = conn),
-    do: uri.host == conn.host and uri.scheme == conn.scheme and uri.port == conn.port
+
+  defp origin_allowed?(:conn, uri, _endpoint, %Plug.Conn{} = conn) do
+    uri.host == conn.host and
+      String.to_atom(uri.scheme) == conn.scheme and
```

Comment:
`String.to_atom` is dangerous We can either do `String.to_existing_atom` or convert `conn.scheme` to a string. Thoughts?

Reply:
I prefer the later one - converting `conn.scheme` to a string by `Atom.to_string/1`.

Is that fine?

Reply:
Yes, please!

Reply:
Done!